flask_jwt_extended/tokens.py

```diff
@@ -125,8 +149,13 @@ def decode_jwt(encoded_token, secret, algorithm, identity_claim_key,
     :param csrf_value: Expected double submit csrf value
     :return: Dictionary containing contents of the JWT
     """
+    # The validation decorator for additional claims must evaluate these! 
+    options = {
```

Comment:
I don't like this, simply checking if these claims exist in the JWT is not enough to verify them. Instead I think we need a way to pass the expected verify options to the underlying pyjwt decode function.